src/main/java/com/thebookcooper/dao/BookInfoDAO.java

```diff
@@ -0,0 +1,76 @@
+package com.thebookcooper.dao;
+
+import com.thebookcooper.model.*;
+import com.thebookcooper.util.DataAccessObject;
+
+import java.sql.*;
+
+
+public class BookInfoDAO extends DataAccessObject<Book> {
+
+    private static final String GET_ONE = "SELECT book_id, title, isbn, publish_date, author, genre " +
+            "FROM book_info WHERE book_id=?";
+
+    private static final String INSERT = "INSERT INTO book_info (title, isbn, publish_date, author, genre) " +
+            "VALUES (?, ?, ?, ?, ?)";
+
+    public BookInfoDAO(Connection connection) {
+        super(connection);
+    }
+
+
+    @Override
+    public Book findById(long id) {
```

Comment:
if no book is found with the given ID, it returns an empty Book object. hsouldnt it make more sense to returna  null?

Reply:
i think either is fine. Either way, we can check no book is found by checking the book object being empty or null. I think its fine to leave it as is